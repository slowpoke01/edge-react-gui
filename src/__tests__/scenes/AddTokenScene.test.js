/* globals describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { AddToken } from '../../components/scenes/AddTokenScene.js'
import { getTheme } from '../../components/services/ThemeContext.js'

describe('Request', () => {
  it('should render with loading props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      route: {
        name: 'addToken',
        params: ['Ethereum', 'ETH', '0x123', '18']
      },
      addNewToken: (walletId, currencyName, currencyCode, contractAddress, denomination, type) => undefined,
      addTokenPending: true,
      currentCustomTokens: [
        {
          currencyName: 'Ethereum',
          currencyCode: 'ETH',
          contractAddress: '1234',
          multiplier: '1000000000',
          denomination: [{ multiplier: '100000000', name: 'BTC' }],
          isVisible: true,
          denominations: [{ multiplier: '100000000', name: 'BTC' }]
        }
      ],
      wallet: {
        id: 'myWallet',
        type: '',
        name: '',
        primaryNativeBalance: '43985',
        nativeBalances: ['ETH'],
        currencyNames: ['Ethtereum'],
        currencyCode: 'ETH',
        isoFiatCurrencyCode: {
          wallet: {
            fiatCurrencyCode: 'USD'
          },
          fiatCurrencyCode: 'USD',
          symbolImage: '₿',
          symbolImageDarkMono: '₿',
          metaTokens: {
            currencyCode: 'BTC',
            currencyName: 'Bitcoin',
            contractAddress: '123123',
            denominations: [],
            symbolImage: '₿'
          },
          enabledTokens: 'ETH',
          receiveAddress: '123123',
          addressLoadingProgress: 11,
          blockHeight: 11
        }
      },
      theme: getTheme()
    }
    const actual = renderer.render(<AddToken {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
