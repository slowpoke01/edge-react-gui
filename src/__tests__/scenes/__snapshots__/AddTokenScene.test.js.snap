// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request should render with loading props 1`] = `
<SceneWrapper
  background="body"
>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 28.091549295774648,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 28.091549295774648,
          "paddingVertical": 22.473239436619718,
        }
      }
    >
      <FormattedText
        style={
          Object {
            "fontSize": 22.473239436619718,
            "textAlign": "center",
          }
        }
      >
        To add a token, please fill out the following form and press "Save" 
      </FormattedText>
    </View>
    <View
      style={
        Object {
          "height": 98.32042253521126,
        }
      }
    >
      <FormField
        autoCapitalize="words"
        autoCorrect={false}
        autoFocus={true}
        keyboardType="default"
        label="Token Name"
        onChangeText={[Function]}
        returnKeyType="done"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 98.32042253521126,
        }
      }
    >
      <FormField
        autoCapitalize="characters"
        autoCorrect={false}
        keyboardType="default"
        label="Token Code"
        maxLength={7}
        onChangeText={[Function]}
        returnKeyType="done"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 98.32042253521126,
        }
      }
    >
      <FormField
        autoCapitalize="none"
        autoCorrect={false}
        keyboardType="default"
        label="Contract Address"
        onChangeText={[Function]}
        returnKeyType="done"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 98.32042253521126,
        }
      }
    >
      <FormField
        autoCapitalize="none"
        autoCorrect={false}
        keyboardType="numeric"
        label="Number of Decimal Places"
        onChangeText={[Function]}
        returnKeyType="done"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "height": 73.03802816901408,
          "justifyContent": "space-between",
          "marginTop": 22.473239436619718,
          "paddingVertical": 5.618309859154929,
        }
      }
    >
      <PrimaryButton
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#0E4B75",
            "borderRadius": 3,
            "flex": 1,
            "marginLeft": 2.8091549295774647,
          }
        }
      >
        <ActivityIndicator
          color="#66EDA8"
        />
      </PrimaryButton>
    </View>
    <View
      style={
        Object {
          "height": 421.3732394366197,
        }
      }
    />
  </ScrollView>
</SceneWrapper>
`;
